app/code/Magento/Config/Block/System/Config/Edit.php

```diff
@@ -41,16 +44,24 @@ class Edit extends \Magento\Backend\Block\Widget
     protected $_configStructure;
 
     /**
+     * @var \Magento\Framework\Serialize\Serializer\Json
+     */
+    private $jsonSerializer;
+
+    /**
      * @param \Magento\Backend\Block\Template\Context $context
      * @param \Magento\Config\Model\Config\Structure $configStructure
      * @param array $data
+     * @param Json|null $jsonSerializer
      */
     public function __construct(
         \Magento\Backend\Block\Template\Context $context,
         \Magento\Config\Model\Config\Structure $configStructure,
-        array $data = []
+        array $data = [],
+        Json $jsonSerializer = null
```

Comment:
As mentioned on https://github.com/magento/magento2/pull/8331#discussion_r99838100 and referenced on [#10342 (comment)](https://github.com/magento/magento2/pull/10342/files#r129688126) , `SerializerInterface` is not guaranteed to return a _JSON_ serializer in the future. And `\Magento\Config\Block\System\Config` uses it in a [method explicitly named `getConfigSearchParamsJson`](https://github.com/liip-forks/magento2/blob/94b423bf985dbd210a5eb13a43ff4dd92a8cc3ca/app/code/Magento/Config/Block/System/Config/Edit.php#L124)